service/src/main/java/org/cbioportal/service/util/MutationMapperUtils.java

```diff
@@ -51,6 +45,48 @@
             filteredFusions = uniqueFusionMap.values().stream().collect(Collectors.toList());
         }
 
+        Set<String> geneSymbolsFromProteinChange = filteredFusions
+            .stream()
+            .filter(fusion -> fusion.getProteinChange() != null && 
+                !fusion.getProteinChange().equalsIgnoreCase("Fusion") && 
+                !fusion.getProteinChange().equalsIgnoreCase("SV"))
+            .flatMap(fusion -> {
+                String proteinChange = fusion.getProteinChange();
+                Matcher matcher = r.matcher(proteinChange);
+                List<String> symbols = new ArrayList<>();
+                if(matcher.matches()) {
+                    if(matcher.group(3) != null) {
+                        symbols.add(matcher.group(1).toUpperCase());
+                        symbols.add(matcher.group(2).toUpperCase());
+                    } else if(matcher.group(2) != null) {
```

Comment:
Can group 2 ever be null? I don't think you need this check.

Reply:
yes, it can be null.